File: Crowd-Source/src/main/java/com/sih/crowdsource/entity/Complaint.java
package com.sih.crowdsource.entity;

import com.sih.crowdsource.enums.ComplaintState;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "complaints")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Complaint {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @Column(length = 1000)
    private String description;

    private String location;

    @Enumerated(EnumType.STRING)
    private ComplaintState status;

    private LocalDateTime createdAt;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User createdBy;

    @ManyToOne
    @JoinColumn(name = "department_id")
    private Department department;

    @OneToMany(mappedBy = "complaint", cascade = CascadeType.ALL)
    private List<Attachment> attachments;
}

File: Flutter_FrontEnd/crowd_source_civic_app/lib/Data/models/complaint.dart
import 'complaint_state.dart';

class Complaint {
  final int id;
  final String title;
  final String description;
  final String location;
  final ComplaintState status;
  final DateTime createdAt;
  final String departmentName;

  Complaint({
    required this.id,
    required this.title,
    required this.description,
    required this.location,
    required this.status,
    required this.createdAt,
    required this.departmentName,
  });

  factory Complaint.fromJson(Map<String, dynamic> json) {
    return Complaint(
      id: json['id'] as int,
      title: json['title'] as String,
      description: json['description'] as String,
      location: json['location'] as String,
      status: ComplaintState.fromString(json['status'] as String),
      createdAt: DateTime.parse(json['createdAt'] as String),
      departmentName: json['departmentName'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'location': location,
      'status': status.name,
      'createdAt': createdAt.toIso8601String(),
      'departmentName': departmentName,
    };
  }
}
